google/resource_compute_target_tcp_proxy_test.go

```diff
@@ -0,0 +1,119 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccComputeTargeTcpProxy_basic(t *testing.T) {
+	target := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+	backend := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+	hc := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeTargetTcpProxyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeTargetTcpProxy_basic1(target, backend, hc),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeTargetTcpProxyExists(
+						"google_compute_target_tcp_proxy.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccComputeTargetTcpProxy_update(t *testing.T) {
+	target := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+	backend := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+	hc := fmt.Sprintf("ttcp-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeTargetTcpProxyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
```

Comment:
How is this testing the update feature? You need at least two `TestStep`. Make sure to test the update with the proxyHeader field (only updateable field) otherwise it will do a Delete/Create operation which won't test the update method.